Sources/WordPressUI/Extensions/UIImageView+Networking.swift

```diff
@@ -56,69 +55,73 @@ public extension UIImageView {
 
     /// Downloads an image and updates the current UIImageView Instance.
     ///
-    /// - Parameters:
-    ///     -   request: The request for the target image
-    ///     -   placeholderImage: Image to be displayed while the actual asset gets downloaded.
-    ///     -   success: Closure to be executed on success.
-    ///     -   failure: Closure to be executed upon failure.
+    /// - parameters:
+    ///    - request: The request for the target image
+    ///    - placeholderImage: Image to be displayed while the actual asset gets downloaded.
+    ///    - success: Closure to be executed on success.
+    ///    - failure: Closure to be executed upon failure.
     ///
-    @objc func downloadImage(usingRequest request: URLRequest, placeholderImage: UIImage? = nil, success: ((UIImage) -> ())? = nil, failure: ((Error?) -> ())? = nil) {
-        cancelImageDownload()
-
-        let handleSuccess = { [weak self] (image: UIImage, url: URL) in
+    ///  - note: The callbacks are executed on the main queue. If you cancel the
+    ///  request, the previous callbacks are guaranteed not to be called: make
+    ///  sure to perform any cleanup before calling this method.
+    @objc func downloadImage(
+        usingRequest request: URLRequest,
+        placeholderImage: UIImage? = nil,
+        success: ((UIImage) -> ())? = nil,
+        failure: ((Error?) -> ())? = nil
+    ) {
+        assert(Thread.isMainThread, "The downloadImage method has to be called from the main thread")
+
+        let context = self.context
+        context.cancel()
+
+        let handleSuccess = { [weak self] (image: UIImage) in
             self?.image = image
             success?(image)
         }
 
         guard let url = request.url else {
-            if let placeholderImage = placeholderImage {
-                image = placeholderImage
+            if let placeholderImage {
+                self.image = placeholderImage
             }
-
             failure?(ImageDownloadError.noURLSpecifiedInRequest)
             return
         }
 
-        if let cachedImage = ImageCache.shared.getImage(forKey: url.absoluteString) {
-            handleSuccess(cachedImage, url)
+        let cacheKey = url.absoluteString
```

Comment:
It's pretty expensive to calculate `absoluteString`, so we are now doing it only once.